pilot/pkg/serviceregistry/kube/controller/serviceexportcontroller.go

```diff
@@ -89,7 +89,7 @@ func (sc *ServiceExportController) Run(stopCh <-chan struct{}) {
 }
 
 func (sc *ServiceExportController) HandleNewService(obj *v1.Service) error {
-	if sc.pushContext.IsServiceClusterLocal(obj) {
+	if sc.environment.PushContext.IsServiceClusterLocal(obj) {
```

Comment:
You don't want to use the `PushContext` here ... instead do something like `sc.environment.Mesh().IsServiceClusterLocal(obj)`, or maybe `sc.environment.IsServiceClusterLocal(obj)`